public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/setQueryValue.ts

```diff
@@ -6,6 +6,11 @@ export function setResource(query: AzureMonitorQuery, resourceURI: string | unde
     azureMonitor: {
       ...query.azureMonitor,
       resourceUri: resourceURI,
+      metricNamespace: undefined,
+      metricName: undefined,
+      aggregation: undefined,
+      timeGrain: '',
+      dimensionFilters: [],
```

Comment:
we have faced several bugs related to this in the past, do you mind adding a unit test here?